src/onnx/parse_gridsample.cpp

```diff
@@ -0,0 +1,432 @@
+        auto weight_index_t = concat_on_first_dim(info, weight_indices);
+        weight_index_t      = info.add_instruction(
+            make_op("reshape", {{"dims", {weight_indices.size(), 3}}}), weight_index_t);
+        for(auto i = 0; i < 4; ++i)
+        {
+            auto indices    = indices_all.at(i);
+            auto validation = validation_all.at(i);
+            auto indices_t  = concat_on_first_dim(info, indices);
+            indices_t       = info.add_instruction(
+                make_op("reshape", {{"dims", {indices_t->get_shape().elements() / 4, 4}}}),
+                indices_t);
+            auto samples      = info.add_instruction(make_op("gathernd"), m_input, indices_t);
+            auto validation_t = concat_on_first_dim(info, validation);
+            samples           = info.add_common_op("where", validation_t, samples, m_zero_l);
+            auto weights =
+                info.add_instruction(make_op("gathernd"), m_corner_weights.at(i), weight_index_t);
+            weighted_corners.push_back(info.add_instruction(make_op("mul"), samples, weights));
+        }
+
+        auto samples = weighted_corners.at(0);
+        std::for_each(std::next(weighted_corners.begin()),
+                      weighted_corners.end(),
+                      [&info, &samples](auto& s) {
+                          samples = info.add_instruction(make_op("add"), samples, s);
+                      });
+        samples = info.add_instruction(
+            make_op("reshape", {{"dims", {m_batch, m_out_height, m_out_width, m_channel}}}),
+            samples);
+        samples =
+            info.add_instruction(make_op("transpose", {{"permutation", {0, 3, 1, 2}}}), samples);
+        samples = info.add_instruction(
+            make_op("convert", {{"target_type", m_input->get_shape().type()}}), samples);
+        return samples;
+    }
+};
+
+struct parse_gridsample : op_parser<parse_gridsample>
+{
+    std::vector<op_desc> operators() const { return {{"GridSample"}}; }
+    instruction_ref parse(const op_desc& /*opd*/,
+                          const onnx_parser& parser,
+                          const onnx_parser::node_info& info,
+                          std::vector<instruction_ref> args) const
+    {
+        bool align_corners = false;
+        // Note: defult mode can be linear or bilinear depending on the onnx version
+        std::string mode         = "linear";
+        std::string padding_mode = "zeros";
+
+        if(contains(info.attributes, "align_corners"))
+        {
+            align_corners = parser.parse_value(info.attributes.at("align_corners")).at<bool>();
+        }
+
+        if(contains(info.attributes, "mode"))
+        {
+            mode = info.attributes.at("mode").s();
+            // Note: can be cubic or bicubic depending on the onnx version
+            if(contains(mode, "cubic"))
+            {
+                MIGRAPHX_THROW("PARSE_GRID_SAMPLE: cubic mode is not supported");
```

Comment:
I've already created one test case for this in `test\onnx\parse`: https://github.com/ROCm/AMDMIGraphX/pull/2909/files#diff-370f8944dad2d8d0077880b181893bd879ecf4830803611ffd33cf09305d67cc
I have no clue why codecov doesn't recognize it.

Reply:
It turned out that the test onnx file was failing with a parse error instead of this. Not sure if it's the cause of the codecov not recognizing. I will update that when we have access to the rack again.